internal/gps/pkgtree/pkgtree.go

```diff
@@ -928,3 +944,42 @@ func uniq(a []string) []string {
 	}
 	return a[:i]
 }
+
+// CreateIgnorePrefixTree takes a set of strings to be ignored and returns a
+// trie consisting of strings prefixed with wildcard ignore suffix (*).
+func CreateIgnorePrefixTree(ig map[string]bool) *radix.Tree {
+	var xt *radix.Tree
```

Comment:
Why not initialize a new tree here instead of checking for nil later on?

Reply:
We had that earlier but then we decided to allocate memory only when we actually get a wildcard suffix. Refer https://github.com/golang/dep/pull/1156#discussion_r138512356